nbgrader_to_canvas/upload_grades.py

```diff
@@ -0,0 +1,379 @@
+        update_status(self._form_nb_assign_name, self._course_id, completion=5, status='Removing Old Feedback')
+        counter = 0
+        auth_header = {'Authorization': 'Bearer ' + self._flask_session['api_key']}
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            for student in gb.students:
+                try:
+                    submission = self.assignment_to_upload.get_submission(self.canvas_students[student.id],include=['submission_comments'])
+                    
+                    for comment in submission.submission_comments:
+                        if comment['comment'] == 'Transferred from datahub.':
+                            response = requests.delete(
+                                "{}api/v1/courses/{}/assignments/{}/submissions/{}/comments/{}".format(
+                                    settings.API_URL, self._course_id, self.assignment_to_upload.id, student.id, comment['id']
+                                ),
+                                headers = auth_header
+                                )
+                except Exception as ex:
+                    raise ex
+                    app.logger.error('Error deleting comments: {}'.format(ex))
+                if counter < self._num_students:
+                    counter += 1
+                    update_status(self._form_nb_assign_name, self._course_id, completion=5+15*counter/self._num_students)
+            
+
+    def _get_feedbacks(self):
+        update_status(self._form_nb_assign_name, self._course_id, completion=20, status='Fetching Feedback')
+        counter = 0
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            for student in gb.students:
+                if not os.path.isdir('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name):
+                    continue
+                for file in os.listdir('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name):
+                    if file.endswith('.html'):
+                        files = {'file': open('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name+'/'+file, 'rb')}
+                        r_json = self._prepare_feedback(student, file)  
+                        response = self._upload_feedback(r_json, files)
+                        r_json = response.json()
+                        if response.status_code >= 300 and response.status_code < 400:
+                            r_json = self._confirm_feedback(r_json) 
+                        self.nbgrader_feedback[student.id] = r_json['id']
+                if counter < self._num_students:
+                    counter += 1
+                    update_status(self._form_nb_assign_name, self._course_id, completion=20+15*counter/self._num_students)
+
```

Comment:
perhaps put the the database code into a repository class and put the http code into another class